scene/2d/cpu_particles_2d.cpp

```diff
@@ -1020,22 +1020,22 @@ void CPUParticles2D::_update_particle_data_buffer() {
 				ptr[6] = 0;
 				ptr[7] = t.elements[2][1];
 
+				Color c = r[idx].color;
+				uint8_t *data8 = (uint8_t *)&ptr[8];
+				data8[0] = CLAMP(c.r * 255.0, 0, 255);
+				data8[1] = CLAMP(c.g * 255.0, 0, 255);
+				data8[2] = CLAMP(c.b * 255.0, 0, 255);
+				data8[3] = CLAMP(c.a * 255.0, 0, 255);
+
+				ptr[9] = r[idx].custom[0];
+				ptr[10] = r[idx].custom[1];
+				ptr[11] = r[idx].custom[2];
+				ptr[12] = r[idx].custom[3];
+
 			} else {
-				zeromem(ptr, sizeof(float) * 8);
+				zeromem(ptr, sizeof(float) * 13);
 			}
 
-			Color c = r[idx].color;
-			uint8_t *data8 = (uint8_t *)&ptr[8];
-			data8[0] = CLAMP(c.r * 255.0, 0, 255);
-			data8[1] = CLAMP(c.g * 255.0, 0, 255);
-			data8[2] = CLAMP(c.b * 255.0, 0, 255);
-			data8[3] = CLAMP(c.a * 255.0, 0, 255);
-
-			ptr[9] = r[idx].custom[0];
-			ptr[10] = r[idx].custom[1];
-			ptr[11] = r[idx].custom[2];
-			ptr[12] = r[idx].custom[3];
-
 			ptr += 13;
```

Comment:
This change may be able to be left out, but I noticed that even in the case of inactive particles, we are still setting the color and custom values, which on the face of it, seems unnecessary. So I moved that into the bracketed if statement above.

In addition, the zeromem is for 8 floats, whereas the structure is 13 floats. Is zeroing 13 more appropriate? 8 may be enough to prevent something being drawn but the other values may be used in calculations on the GPU. This may not be a problem once the Particles are zeroed in `set_amount`, the only danger as I see it is NaNs in here.

There is another loop in `NOTIFICATION_TRANSFORM_CHANGED` where this may also be relevant.